### PR TITLE
Features/sidebar

### PR DESCRIPTION
**Pull request by scokljat**
## Changes 

- Add animated and responsive sidebar with style. 
- Use a custom component  router-link to create links. 

## Reference

https://trello.com/c/JxLA7Sge/32-task-features-sidebar

### CODE REVIEW

File: src/components/TheSidebar.vue [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  position: fixed;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  transition: 0.3s ease;
+  display: flex;
+  flex-direction: column;
+}
+
+.sidebarlink {
```
Comment (josip2312) [inline src/components/TheSidebar.vue:50]: This class makes no semantic sense, it is the wrapper of sidebar links, so name it appropriately

File: src/components/TheSidebar.vue [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  left: 1.25rem;
+  top: 0.375rem;
+  padding: 0.3125rem;
+  transition: 0.2s linear;
+  margin-bottom: 1.25rem;
+  font-size: 1.5rem;
+  color: var(--text-primary-1);
+}
+
+.collapsed {
```
Comment (josip2312) [inline src/components/TheSidebar.vue:87]: This would make sense to be in the `sidebar` selector

File: src/components/TheSidebar.vue [hunk 3]
```diff
@@ -0,0 +1,104 @@
+<template>
+  <button class="collapse-icon" @click="toggleSidebar">
+    <i class="fa-solid fa-bars"></i>
+  </button>
+
+  <div class="sidebar" :class="{ collapsed: collapsed }">
```
Comment (josip2312) [inline src/components/TheSidebar.vue:6]:
```suggestion
  <div class="sidebar" :class="{ collapsed }">
```

**Review by josip2312 — CHANGES_REQUESTED** [review]

File: src/components/TheSidebar.vue [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    <div class="sidebarlink">
```
Comment (thedev966) [inline src/components/TheSidebar.vue:7]: You may want to change the class name into "sidebar_link" or "sidebarLInk" depending on what convention you follow.

File: src/components/TheSidebar.vue [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  color: var(--text-primary-1);
+}
+
+.collapse-icon {
+  position: fixed;
+  z-index: 100;
+  left: 1.25rem;
+  top: 0.375rem;
+  padding: 0.3125rem;
+  transition: 0.2s linear;
```
Comment (thedev966) [inline src/components/TheSidebar.vue:81]: Maybe it's better to option to use "ease", "ease-in" transition style instead of "linear"?

**Review by thedev966 — APPROVED** [review]

**Comment by josip2312** [comment]
Two things to address 
- have you agreed on a convention for class naming
- when changing routes on desktop, you are unnecessarily calling the `toggleSidebar` method on each click

**Review by josip2312 — APPROVED** [review]

**Pull request merged**